R/functions_degs.R

```diff
@@ -89,13 +100,53 @@ DegsAvgData = function(sc, genes) {
   return(avg_data)
 }
 
-#' Write differentially expressed genes to file
+
+#' Compute average gene expression data for a set of cells and a set of genes.
 #' 
-#' @param degs Result table of the "Seurat::FindAllMarkers" function (requires column "cluster"). The table is split by the column "cluster", and each split table is written into a separate Excel tab. 
-#' @param annot_ensembl Ensembl annotation for all genes
-#' @param file Output file name 
-DegsWriteToFile = function(degs, annot_ensembl, file) {
+#' @param object Seurat assay object
+#' @param cells Cells to be used. Can be NULL where all cells will be used
+#' @param genes Gene list for which average data are to be extracted. Can be NULL where all genes will be calculated
+#' @param slot Slot to be used (data). Can be 'counts' or 'data' or both (vector).
+#' @return A table with average data for each gene
+DegsAvgData = function(object, cells=NULL, genes=NULL, slot="data") {
```

Comment:
This is because for the markers we calculate a mean of raw and normalized data for each identity class which blows up the table. For standard degs, we just compute a mean for group1 and one for group2.

Of course the two function could be combined but this may be a minor issue.

Reply:
Ok, if you didn't duplicate code.

Reply:
Then I leave it